astropy/cosmology/funcs/tests/test_funcs.py

```diff
@@ -299,15 +308,23 @@ def test_z_at_value_bracketed(method):
             rtol=1e-6,
         )
 
-    with pytest.raises(core.CosmologyError):
-        with pytest.warns(AstropyUserWarning, match=r"fval is not bracketed"):
-            z_at_value(
-                cosmo.angular_diameter_distance,
-                1500 * u.Mpc,
-                method=method,
-                bracket=(3.9, 5.0),
-                zmin=4.0,
-            )
+    if not PYTEST_LT_8_0 and method == "Bounded":
+        ctx = pytest.warns(
+            UserWarning, match="Option 'bracket' is ignored by method Bounded"
+        )
```

Comment:
If given a distinct name, this ctx can be defined instead around line 204 and used in line 215.

Reply:
I don't think so? Its value is highly depended on input and pytest version here.